### PR TITLE
[SPARK-37354][K8S][TESTS] Make the Java version installed on the container image used by the K8s integration tests with SBT configurable

### PR DESCRIPTION
**Pull request by sarutak**
### What changes were proposed in this pull request?

This PR fixes an issue that the Java version installed on the container image used by K8s integration tests are not configurable in case the tests are launched by SBT.

`bin/docker-image-tool.sh` supports `-b java_image_tag=<java image tag>` option to the Java version installed on the container image configurable.
But `project/SparkBuild.scala` doesn't provide the way to pass the option so we can't change the Java version if the tests are launched by SBT.

To allow to change the version, this PR changes `project/SparkBuild.scala` to recognized the environment variable `JAVA_IMAGE_TAG`, which is the same manner that `resource-managers/kubernetes/integration-tests/scripts/setup-integration-test-env.sh` does.

### Why are the changes needed?

The K8s integration tests should be launched by any supported Java versions.

### Does this PR introduce _any_ user-facing change?

No.

### How was this patch tested?

Confirmed that the Java version installed on the container image.

In case `JAVA_IMAGE_TAG` is not set:
```
# Run the integration tests to create the container image
$ build/sbt -Phive -Pkubernetes  -Pkubernetes-integration-tests package "kubernetes-integration-tests/test"  

# Create and login the container.
docker run -it <image>:<tag> /bin/bash

# Confirm the version in the container.
185@4023f49aa0b0:/opt/spark/work-dir$ java -version
openjdk version "1.8.0_312"
OpenJDK Runtime Environment (build 1.8.0_312-b07)
OpenJDK 64-Bit Server VM (build 25.312-b07, mixed mode)
```

In case `JAVA_IMAGE_TAG` is set:
```
# Run the integration tests to create the container image
$ JAVA_IMAGE_TAG=11-jre-slim build/sbt -Phive -Pkubernetes  -Pkubernetes-integration-tests package "kubernetes-integration-tests/test"  

# Create and login the container.
docker run -it <image>:<tag> /bin/bash

# Confirm the version in the container.
185@fdf2d9d5ee26:/opt/spark/work-dir$ java -version
openjdk version "11.0.13" 2021-10-19
OpenJDK Runtime Environment 18.9 (build 11.0.13+8)
OpenJDK 64-Bit Server VM 18.9 (build 11.0.13+8, mixed mode, sharing)
```

### CODE REVIEW

File: project/SparkBuild.scala [hunk 1]
```diff
@@ -643,6 +643,7 @@ object KubernetesIntegrationTests {
           "-t", imageTag.value,
           "-p", s"$bindingsDir/python/Dockerfile",
           "-R", s"$bindingsDir/R/Dockerfile",
+          "-b", s"java_image_tag=${sys.env.getOrElse("JAVA_IMAGE_TAG", "8-jre-slim")}",
```
Comment (sarutak) [inline project/SparkBuild.scala:646]: The default Java version is 8 if the K8s integration tests are launched by Maven so this PR complies with that.

**Comment by holdenk** [comment]
LGTM, the K8s failures seem unrelated I've got a follow up PR for the decom failure - https://github.com/apache/spark/pull/34636

**Comment by sarutak** [comment]
I'll merge this as I have one LGTM from @holdenk . But I'll rebase before merge,  just in case.

**Comment by sarutak** [comment]
Merging to `master`.
I think the failed Kubernetes integration tests are not related to this change and they are flaky ones. I saw those tests failed in another PR.
https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder-K8s/49651/console

Thank you @holdenk for reviewing !

**Pull request closed without merging**